geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/serialization/AvroSerialization.scala

```diff
@@ -108,5 +109,23 @@ class AvroReader extends AbstractReader[Decoder] {
     decoder.readArrayStart.asInstanceOf[Int]
   }
 
+  // Avro specific reading method...see Decoder javadoc
+  override def readGenericMap(version: Version): DatumReader[Decoder, java.util.Map[AnyRef, AnyRef]] = (reader) => {
```

Comment:
yeah for reals...i believe its due to the fact that deep down inside you can fetch blocks of binary data at a time?